torch/utils/data/dataset.py

```diff
@@ -251,6 +251,39 @@ def __getitem__(self, index):
             return {k: dataset[index] for k, dataset in self.datasets.items()}
         return tuple(dataset[index] for dataset in self.datasets)
 
+    def __getitems__(self, indices: list):
+        # add batched sampling support when parent datasets supports it.
+        if isinstance(self.datasets, dict):
+            dict_batch: List[T_dict] = [{} for _ in indices]
+            for k, dataset in self.datasets.items():
+                if callable(getattr(dataset, "__getitems__", None)):
+                    items = dataset.__getitems__(indices)  # type: ignore[attr-defined]
+                    if len(items) != len(indices):
+                        raise ValueError("Nested dataset's output size mismatch."
+                                         f" Expected {len(indices)}, got {len(items)}")
```

Comment:
Can you pls add an unit test to validate the `ValueError` is raised as expectation?